beater/cloudbeat.go

```diff
@@ -153,9 +155,35 @@ func (bt *cloudbeat) Run(b *beat.Beat) error {
 
 		case update := <-bt.configUpdates:
 			if err := bt.config.Update(update); err != nil {
-				logp.L().Errorf("could not update cloudbeat config: %v", err)
+				logp.L().Errorf("Could not update cloudbeat config: %v", err)
+			}
+
+			policies, err := csppolicies.CISKubernetes()
+			if err != nil {
+				logp.L().Errorf("Could not load CIS Kubernetes policies: %v", err)
+			}
+
+			if len(bt.config.Streams) == 0 {
+				logp.L().Infof("Did not receive any input stream, skipping.")
+				break
+			}
+
+			// TODO(yashtewari): Figure out the scenarios in which the integration sends
+			// multiple input streams. Since only one instance of our integration is allowed per
+			// agent policy, is it even possible that multiple input streams are received?
```

Comment:
As far as I understand it's up to us to define what is being sent. We could as well change the Stream to be a singular instead of an array. Aren't we?

Reply:
I didn't find a way for it to be singular. On the integration end, we just define what to send, and the default structure is that it comes as one input in a list of streams. This would make sense to accommodate a wider variety of integration use cases.